File: frontend/src/app/audit/audit.service.ts
import { API_BASE_URL } from '../api.config';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
export interface AuditLog { id:number; username:string; action:string; resourceType:string; resourceId:string|null; details:string; createdAt:string; }
@Injectable({providedIn:'root'}) export class AuditService { private readonly apiUrl=`${API_BASE_URL}/api/audit-logs`; constructor(private readonly http:HttpClient){} getAll():Observable<AuditLog[]>{return this.http.get<AuditLog[]>(this.apiUrl);} }

File: backend/src/main/java/com/opstrack/audit/AuditService.java
package com.opstrack.audit;

import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class AuditService {
    private final AuditLogRepository repository;
    public AuditService(AuditLogRepository repository) { this.repository = repository; }
    public AuditLog record(String username, String action, String resourceType, Object resourceId, String details) {
        return repository.save(new AuditLog(username, action, resourceType,
                resourceId == null ? null : resourceId.toString(), details));
    }
    public List<AuditLog> getAll() { return repository.findAll(); }
}
